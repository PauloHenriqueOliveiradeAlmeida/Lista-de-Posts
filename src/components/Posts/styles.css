.posts {
    padding: 5% 0px;
    margin: auto;
    display: grid;
    grid-template-columns: repeat(auto-fill, minmax(300px, 6fr));
    gap: 1.4rem;
}