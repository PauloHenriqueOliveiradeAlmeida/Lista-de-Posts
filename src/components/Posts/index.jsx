import P from "prop-types";
import "./styles.css";
import PostCard from "../PostCard";

function Posts({ posts }) {

    return (
        <div className="posts">
            {posts.map(post => (
                <PostCard post={post} key={post.id} />
            ))
            }
        </div>
    )

}


export default Posts;


Posts.propTypes = {
	posts: P.arrayOf(
		P.shape({
			photo: P.string.isRequired,
			title: P.string.isRequired,
			body: P.string.isRequired
		})
	)
}
