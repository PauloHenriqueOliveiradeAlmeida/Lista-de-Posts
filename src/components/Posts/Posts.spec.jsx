import { render, screen } from "@testing-library/react";
import Posts from ".";

const postsMock = [
    {
        id: 1,
        title: "post title",
        body: "post description",
        photo: "image.png"
    },
    {
        id: 2,
        title: "2 post title",
        body: "2 post description",
        photo: "2image.png"
    },
    {
        id: 3,
        title: "3 post title",
        body: "3 post description",
        photo: "3image.png"
    }
];

describe("<Posts />", () => {
    it("should render Posts component", () => {
        render(<Posts posts={postsMock}/>);

        expect(screen.getAllByRole("img", {name: /title/i})).toHaveLength(3);
        expect(screen.getAllByRole("heading", {name: /title/i})).toHaveLength(3);
        expect(screen.getAllByText(/title/i)).toHaveLength(3);

    });

    it("should match snapshot", () => {
        const {container} = render(<Posts posts={postsMock}/>);

        expect(container.firstChild).toMatchSnapshot();
    });
})