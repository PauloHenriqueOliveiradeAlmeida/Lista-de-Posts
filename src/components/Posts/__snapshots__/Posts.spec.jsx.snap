// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="post title"
      src="image.png"
    />
    <div
      class="post-content"
    >
      <h2>
        post title
      </h2>
      <hr />
      <p>
        post description
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="2 post title"
      src="2image.png"
    />
    <div
      class="post-content"
    >
      <h2>
        2 post title
      </h2>
      <hr />
      <p>
        2 post description
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="3 post title"
      src="3image.png"
    />
    <div
      class="post-content"
    >
      <h2>
        3 post title
      </h2>
      <hr />
      <p>
        3 post description
      </p>
    </div>
  </div>
</div>
`;
